src/notebooks_service.py

```diff
@@ -95,10 +97,28 @@ def health():
 @authenticated
 def whoami(user):
     """Return information about the authenticated user."""
-    return Response(
-        json.dumps(user, indent=1, sort_keys=True),
-        mimetype='application/json',
+    headers = {auth.auth_header_name: 'token {0}'.format(auth.api_token)}
+    info = json.loads(
+        requests.request(
+            'GET',
+            '{prefix}/users/{user[name]}'.format(
+                prefix=auth.api_url, user=user
+            ),
+            headers=headers
+        ).text
     )
+    return jsonify(info)
+
+
+# @authenticated
+@app.route(urljoin(SERVICE_PREFIX, 'ui'), defaults={'path': ''})
+@app.route(urljoin(SERVICE_PREFIX, 'ui/'), defaults={'path': ''})
+@app.route(urljoin(SERVICE_PREFIX, 'ui/<path:path>'), methods=['GET'])
+def ui(path):
+    """Route for serving a UI for managing running servers."""
+    if not path:
+        path = "index.html"
+    return send_from_directory('ui/build/', path)
```

Comment:
The ui is a set of static assets, so they should be served with `url_for('static', filename=...)`